SAK-40287 Library: Made the footer positioning more efficient on every page load

fostersdesign opened the pull request:
https://jira.sakaiproject.org/browse/SAK-40287

Every time a page in Sakai loads, there is a JavaScript function that runs to resize the page and push the footer to the bottom. This function produces page jitter as the page is being resized and the footer is moved to the bottom. This makes every page seem slow or laggy.

This Javascript function has been replaced with CSS that positions the footer properly on page load and eliminates the page jitter.

I have also cleaned up scrolling in Resources and in the mobile view by reducing excess page length caused by the hidden tool menu still taking up room.

jonespm commented:
Have you tested this with really long pages (like lessons pages with lots of content)?

Since lessons pages embed other tools as iframes still,testing with (for instance) a samigo assessment delivered all on one page that is long would be a good test for this.

Thanks

fostersdesign commented:
Yes, I tested my change with pages and tools of a variety of different lengths, and sites with different lengths of tool menus. 

Yes, I tested my change with Lessons containing text, links, subpages, student content pages, embedded YouTube videos, and linked quizzes. Long quizzes with all questions on one page shows in an iframe with a set height, so double scrollbars appear in the Lesson. This is the same before and after this change.

bjones86 commented on library/src/morpheus-master/sass/base/_responsive.scss:
```diff
@@ -1,7 +1,14 @@
 body{
-	@media #{$phone}{
+	@media #{$phone}
+	{
```
This file does not use Allman style braces. Can we keep it consistent?

jonespm replied:
Yeah, I agree. It had K&R style (which I personally like over Allman). I think Allman and K&R are used inconsistently throughout Sakai, but nice to at least keep the file to the same style.

bjones86 replied:
Yeah everyone has their favourite. Personally I prefer Allman, but my rule of thumb when working in Sakai is to 'keep it consistent': I try my best to use whatever styles are already most prevalent in the given file.

It's the mixing of styles and whitespace that makes reading some Sakai code a nightmare...

bjones86 commented on library/src/morpheus-master/sass/modules/_main.scss:
```diff
@@ -20,7 +20,11 @@
-	> .#{$namespace}pagebody{
+	
+	> .#{$namespace}pagebody
+	{
```
Same here

bjones86 commented on library/src/morpheus-master/sass/modules/_toolmenu.scss:
```diff
@@ -27,13 +27,8 @@ body.is-logged-out{
-		}
-
+	body.#{$namespace}toolMenu-collapsed
+	{
```
And here

fostersdesign commented:
PR code updated to reflect comments from @bjones86

fostersdesign commented:
I've just uploaded a quick video to the JIRA ticket showing the before and after of my changes by comparing a few tools from the Home site on nightly master (the "before") during the first half of the video and my local host with the changes (the "after") during the second half.

You should notice the footer is positioned after the content and then jumps to the bottom of the page when the JavaScript re-positions it (the before) compared to the footer already being at the bottom of the page when the page loads (the after).

Also notice that the after state contains no scroll bars because the page is the exact height of the screen (when the content is less than the screen height).

fostersdesign commented:
For this fix, I had to add a magic number of 1px to the height calculation because I couldn't find where the 1px was coming from. This bothered me. I have finally found it: it was coming from the height of the #Mrphs-viewport-helpers children.

Since this PR was merged last night, I created SAK-40294. I've removed the 1px height from these children and removed the magic number from my #toolMenuWrap calculation.

I have submitted #5779 just now to remove the pixel. @jonespm and @bjones86, would you mind reviewing #5779  please? Thanks.
